upgrade/util.go

```diff
@@ -47,6 +49,8 @@ type ProviderRepo struct {
 	// The upstream version we are upgrading from.  Because not all upstream providers
 	// are go module compliment, we might not be able to always resolve this version.
 	currentUpstreamVersion *semver.Version
+
+	upstreamName string
```

Comment:
Is this different than `Context.UpstreamProviderName`? Can you add a doc string explaining what it means?

Reply:
They are the same, but not necessary to keep track of it in `ProviderRepo`. Simpler to store it in the context only so I'll remove